Karla Grajales | Module-Data-Groups | Sprint-3   Quote Generator App

Grajales-K opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

I’ve learned to manipulate the DOM by creating and appending elements like buttons and divs. I implemented a random quote generator, handled events to display new quotes and control auto-quotes, and added background sound functionality with the Audio mp3, enriching the user experience. 🔊

## Questions

Ask any questions you have for your reviewer.

cjyuan reviewed (commented):
When I loaded the app using Live Server on my computer, I see these errors in the browser's JS console:

![image](https://github.com/user-attachments/assets/54ea7bbe-51ae-4926-a017-2720aeddb762)

The URL of the page (as shown in my browser) is `http://127.0.0.1:5500/index.html`.

Can you fix these two bugs?

cjyuan commented:
Chrome browser also shows this error:
Uncaught (in promise) NotAllowedError: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD

This is also the first time I am aware of it. Just thought you might be interested to know.
You do not have to fix it. If you want to fix the issue, you can do it after you have completed this PR first.

Grajales-K commented:
> Chrome browser also shows this error: Uncaught (in promise) NotAllowedError: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> 
> This is also the first time I am aware of it. Just thought you might be interested to know. You do not have to fix it. If you want to fix the issue, you can do it after you have completed this PR first.


Thank you for the article @cjyuan I saw yesterday that the audio needs to have a displayed button where to mute or replay. Read more thank you for sharing this. I have to look into it.

Grajales-K commented:
> When I loaded the app using Live Server on my computer, I see these errors in the browser's JS console:
> 
> ![image](https://private-user-images.githubusercontent.com/7696139/404660491-54ea7bbe-51ae-4926-a017-2720aeddb762.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczMDI0OTIsIm5iZiI6MTczNzMwMjE5MiwicGF0aCI6Ii83Njk2MTM5LzQwNDY2MDQ5MS01NGVhN2JiZS01MWFlLTQ5MjYtYTAxNy0yNzIwYWVkZGI3NjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTlUMTU1NjMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjE5MGVjYTFlMjZkZGEyY2EyMmVjMTZiNDE0NjYzNmJjNjAxYjFkZmIyYTc2NTFhMTIzNjNmNjM5NTljZmUxOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.LEHGyhydAA6hmfiq_e26ysRNzS_5efDxfsrCTjyx4vk)
> 
> The URL of the page (as shown in my browser) is `http://127.0.0.1:5500/index.html`.
> 
> Can you fix these two bugs?

Thank you, @cjyuan. I removed the script in the HTML to fix the error. I moved the media to the assets folder relative to the root directory. I can not see the error with the image on my side. Thank you for your quick response.

Grajales-K commented:
> > Chrome browser also shows this error: Uncaught (in promise) NotAllowedError: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
> > This is also the first time I am aware of it. Just thought you might be interested to know. You do not have to fix it. If you want to fix the issue, you can do it after you have completed this PR first.
> 
> Thank you for the article @cjyuan I saw yesterday that the audio needs to have a displayed button where to mute or replay. Read more thank you for sharing this. I have to look into it.

![Uploading Screenshot 2025-01-19 at 16.08.19.png…]()

Grajales-K commented:
<img width="1440" alt="Screenshot 2025-01-19 at 16 08 19" src="https://github.com/user-attachments/assets/fb2aaa4e-3048-4dff-b6d5-7d54eab9658b" />

cjyuan commented on Sprint-3/quote-generator/style.css:
```diff
@@ -1 +1,75 @@
 /** Write your CSS in here **/
+body {
+    background-image: url("/Sprint-3/quote-generator/stars.webp");
```
Instead of using an absolute path, it is better to use a path "relative to `style.css`".

cjyuan reviewed (commented):
I just suggested a way to fix the image URL problem. Feel free to mark this PR as completed anytime.
